src/Compiler/Symbols/FSharpDiagnostic.fs

```diff
@@ -71,15 +166,48 @@ type FSharpDiagnostic(m: range, severity: FSharpDiagnosticSeverity, message: str
         sprintf "%s (%d,%d)-(%d,%d) %s %s %s" fileName s.Line (s.Column + 1) e.Line (e.Column + 1) subcategory severity message
 
     /// Decompose a warning or error into parts: position, severity, message, error number
-    static member CreateFromException(diagnostic: PhasedDiagnostic, severity, fallbackRange: range, suggestNames: bool, flatErrors: bool) =
-        let m = match diagnostic.Range with Some m -> m | None -> fallbackRange 
-        let msg = diagnostic.FormatCore(flatErrors, suggestNames)
+    static member CreateFromException(diagnostic: PhasedDiagnostic, severity, fallbackRange: range, suggestNames: bool, flatErrors: bool, symbolEnv: SymbolEnv option) =
```

Comment:
Same for the method, probably makes more sense to create a second overload.

Reply:
we need to write more maintenance code: a common internal function that will take all possible parameters and a couple of overloads. Perhaps it makes sense to make this parameter more optional (with ?) ?